Compilation failure (IdentityHashMap) on recent Scala 2.13.2 nightly

at e.g. https://scala-ci.typesafe.com/view/scala-2.13.x/job/scala-2.13.x-integrate-community-build/2658/artifact/logs/scala-js-build.log we see:

```
[scala-js] [error] /home/jenkins/workspace/scala-2.13.x-integrate-community-build/target-0.9.16/project-builds/scala-js-7654802acae1122fadcbca880b40e51c7700e137/javalib/src/main/scala/java/util/IdentityHashMap.scala:43: error: ambiguous reference to overloaded definition,
[scala-js] [error] both constructor IdentityHashMap in class IdentityHashMap of type (initialMap: java.util.Map[_ <: K, _ <: V])java.util.IdentityHashMap[K,V]
[scala-js] [error] and  constructor IdentityHashMap in class IdentityHashMap of type (inner: java.util.HashMap[java.util.IdentityHashMap.IdentityBox[K],V])java.util.IdentityHashMap[K,V]
[scala-js] [error] match argument types (java.util.HashMap[java.util.IdentityHashMap.IdentityBox[K],V]) and expected result type AnyRef
[scala-js] [error]     new IdentityHashMap(
[scala-js] [error]     ^
```

Scala version is 2.13.2-bin-59dec61

--- a/javalib/src/main/scala/java/util/IdentityHashMap.scala
+++ b/javalib/src/main/scala/java/util/IdentityHashMap.scala
@@ -18,7 +18,11 @@ import scala.annotation.tailrec
 
 import ScalaOps._
 
-class IdentityHashMap[K, V](inner: HashMap[IdentityHashMap.IdentityBox[K], V])
+/* The additional `internal` parameter works around
+ * https://github.com/scala/bug/issues/11755
+ */
+class IdentityHashMap[K, V] private (
+    inner: HashMap[IdentityHashMap.IdentityBox[K], V], internal: Boolean)
     extends AbstractMap[K, V] with Map[K, V] with Serializable with Cloneable {
   self =>
 
@@ -26,7 +30,7 @@ class IdentityHashMap[K, V](inner: HashMap[IdentityHashMap.IdentityBox[K], V])
 
   def this(expectedMaxSize: Int) = {
     this(new HashMap[IdentityHashMap.IdentityBox[K], V](
-        expectedMaxSize, HashMap.DEFAULT_LOAD_FACTOR))
+        expectedMaxSize, HashMap.DEFAULT_LOAD_FACTOR), internal = true)
   }
 
   def this() =
@@ -41,7 +45,7 @@ class IdentityHashMap[K, V](inner: HashMap[IdentityHashMap.IdentityBox[K], V])
 
   override def clone(): AnyRef = {
     new IdentityHashMap(
-        inner.clone().asInstanceOf[HashMap[IdentityBox[K], V]])
+        inner.clone().asInstanceOf[HashMap[IdentityBox[K], V]], internal = true)
   }
 
   override def containsKey(key: Any): Boolean =

